File: Modules/Common/Equipments/SIE.Web.Fixtures/Models/Commands/SelProductDetailCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.Fixtures.Models;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.Fixtures.Models.Commands
{
    /// <summary>
    /// 添加产品清单
    /// </summary>
    [JsCommand("SIE.Web.Fixtures.Models.Commands.SelProductDetailCommand")]
    public class SelProductDetailCommand : ViewCommand
    {
        /// <summary>
        /// 执行添加
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = ClientEntities.Find(args.Type);
            var savedData = RF.Find(meta.EntityType).NewList();
            var productDetails = args.Data.ToJsonObject<List<FixtureEncodeProductDetail>>();
            Check.NotNullOrEmpty(productDetails, nameof(productDetails));
            if (null == productDetails || productDetails.Count == 0)
                throw new ValidationException("产品清单不能为空!".L10N());
            foreach (var item in productDetails)
            {
                var equipDetail = new FixtureEncodeProductDetail();
                equipDetail.FixtureEncodeId = item.FixtureEncodeId;
                equipDetail.ItemId = item.ItemId;
                savedData.Add(equipDetail);
            }
            RF.Save(savedData);
            return true;
        }
    }
}

File: Modules/Common/Equipments/SIE.Web.Fixtures/Models/Commands/SelProductDetailCommand.js
SIE.defineCommand('SIE.Web.Fixtures.Models.Commands.SelProductDetailCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'ItemId', targetClassName: 'SIE.Items.Item' }
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    save: function (win) {
        /// <summary>
        /// 保存选择的操作列表。
        /// </summary>
        var me = this;
        var indata = {};
        var selections = me._targetSelectItems;
        if (selections && selections.items.length > 0) {
            var items = [];
            SIE.each(selections.items, function (sel) {
                var id = sel.getId();
                if (me._sourceViewSelectItems.indexOf(id) === -1) {
                    var item = { FixtureEncodeId: me._sourceId, ItemId: id };
                    items.push(item);
                }
            });
            if (items.length > 0) {
                indata = items;
                me._targetView.execute({
                    data: indata,
                    success: function (res) {
                        win.close();
                        me._ownerView.loadChildData(true);
                    }
                }, me._ownerView);
                return true;
            }
        }
        Ext.Msg.alert('提示'.t(), '没有可提交的数据'.t());
    }
});